server/concepts/Tournament.ts

```diff
@@ -85,7 +84,7 @@ export default class Tournament implements TournamentWithoutMS {
           const childTeamsAmountInB = Math.floor(temp);
 
           if (childTeamsAmountInA === 1) {
-            matches[2 ** i - 1 + ii].teamA = teamsToAssign.shift();
+            matches[2 ** i - 1 + ii].teamA = teamsToAssign.shift() || null;
```

Comment:
Hmm it can not be null. I'm sure .shift() return Team object. So I think a good option is use  non-null assertion operator `!` instead, `| null`;

Reply:
it can return undefinied

Reply:
Yes, but i'm sure it will return object, not undefined. It follows from the calculation

Reply:
pushed as you suggested